components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/scope/AbstractRoleBasedScopeIssuer.java

```diff
@@ -0,0 +1,356 @@
+     */
+    public String[] getRolesFromAssertion(Assertion assertion) {
+
+        List<String> roles = new ArrayList<>();
+        String roleClaim = getRoleClaim();
+        List<AttributeStatement> attributeStatementList = assertion.getAttributeStatements();
+
+        if (attributeStatementList != null) {
+            for (AttributeStatement statement : attributeStatementList) {
+                List<Attribute> attributesList = statement.getAttributes();
+                for (Attribute attribute : attributesList) {
+                    String attributeName = attribute.getName();
+                    if (roleClaim.equals(attributeName)) {
+                        List<XMLObject> attributeValues = attribute.getAttributeValues();
+                        if (attributeValues != null && attributeValues.size() == 1) {
+                            String attributeValueString = getAttributeValue(attributeValues.get(0));
+                            String multiAttributeSeparator = getAttributeSeparator();
+                            String[] attributeValuesArray = attributeValueString.split(multiAttributeSeparator);
+                            if (log.isDebugEnabled()) {
+                                log.debug("Adding attributes for Assertion: " + assertion + " AttributeName : "
+                                        + attributeName + ", AttributeValue : " + Arrays
+                                        .toString(attributeValuesArray));
+                            }
+                            roles.addAll(Arrays.asList(attributeValuesArray));
+                        } else if (attributeValues != null && attributeValues.size() > 1) {
+                            for (XMLObject attributeValue : attributeValues) {
+                                String attributeValueString = getAttributeValue(attributeValue);
+                                if (log.isDebugEnabled()) {
+                                    log.debug("Adding attributes for Assertion: " + assertion + " AttributeName : "
+                                            + attributeName + ", AttributeValue : " + attributeValue);
+                                }
+                                roles.add(attributeValueString);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        if (log.isDebugEnabled()) {
+            log.debug("Role list found for assertion: " + assertion + ", roles: " + roles);
+        }
+        return roles.toArray(new String[0]);
+    }
+
+
+    private String getAttributeValue(XMLObject attributeValue) {
+
+        if (attributeValue == null) {
+            return null;
+        } else if (attributeValue instanceof XSString) {
+            return getStringAttributeValue((XSString) attributeValue);
+        } else if (attributeValue instanceof XSAnyImpl) {
+            return getAnyAttributeValue((XSAnyImpl) attributeValue);
+        } else {
```

Comment:
else is redundant

Reply:
fixed https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2084/commits/ad343f6bb0ec0f4978d6e599ac94b1835dab76de